Modules/_decimal/_decimal.c

```diff
@@ -3311,8 +3357,43 @@ dec_format(PyObject *dec, PyObject *args)
         }
     }
 
+    if (no_neg_0 && mpd_isnegative(mpd) && !mpd_isspecial(mpd)) {
```

Comment:
There's a corner case that looks as though it's not being handled correctly here, though I can't put my finger on exactly where it's going wrong:
```python
>>> format(Decimal("-0.0000"), "f")
'-0.0000'
>>> format(Decimal("-0.0000"), "zf")  # expect '0.0000'
'0'
```

Reply:
thank you-- I was incorrectly rounding even when `prec` was < 0